tester/src/test/scala/spinal/tester/scalatest/FormalArbiterTester.scala

```diff
@@ -0,0 +1,455 @@
+    FormalConfig
+      .withBMC(20)
+      .withProve(20)
+      .withCover(20)
+      // .withDebug
+      .doVerify(new Component {
+        val portCount = 5
+        val dataType = Bits(8 bits)
+
+        val dut = FormalDut(
+          new StreamArbiter(dataType, portCount)(StreamArbiter.Arbitration.sequentialOrder, StreamArbiter.Lock.none)
+        )
+        val reset = ClockDomain.current.isResetActive
+
+        assumeInitial(reset)
+
+        val select = UInt(log2Up(portCount) bit)
+        val selectOH = Bits(portCount bit)
+        val inputs = Vec(slave(Stream(dataType)), portCount)
+        val output = master(Stream(dataType))
+
+        when(reset || past(reset)) {
+          for (i <- 0 until portCount) {
+            assume(inputs(i).valid === False)
+          }
+        }
+
+        select := dut.io.chosen
+        selectOH := dut.io.chosenOH
+        output << dut.io.output
+
+        for (i <- 0 until portCount) {
+          inputs(i) >> dut.io.inputs(i)
+          inputs(i).formalAssumesSlave()
+        }
+        output.formalCovers(2)
+        output.formalAssertsMaster()
+
+        assert(select < portCount)
+        assert(select === OHToUInt(selectOH))
+        assert(output.payload === inputs(select).payload)
+
+        val d1 = anyconst(UInt(log2Up(portCount) bit))
+        assume(d1 < portCount)
+        val d2 = UInt(log2Up(portCount) bit)
+        d2 := (d1 + 1) % portCount
+
+        val cntSeqCheck = changed(select) && (select === d2)
+        cover(cntSeqCheck)
+        when(cntSeqCheck) {
+          assert(past(select) === d1)
+        }
+      })
+  }
+
+  test("Arbiter-lowerfirst-none-verify") {
+    FormalConfig
+      .withBMC(20)
+      .withProve(20)
+      .withCover(20)
+      // .withDebug
+      .doVerify(new Component {
+        val portCount = 5
+        val dataType = Bits(8 bits)
+
+        val dut = FormalDut(
+          new StreamArbiter(dataType, portCount)(StreamArbiter.Arbitration.lowerFirst, StreamArbiter.Lock.none)
+        )
+        val reset = ClockDomain.current.isResetActive
+
+        assumeInitial(reset)
+
+        val select = UInt(log2Up(portCount) bit)
+        val selectOH = Bits(portCount bit)
+        val inputs = Vec(slave(Stream(dataType)), portCount)
+        val output = master(Stream(dataType))
+
+        select := dut.io.chosen
+        selectOH := dut.io.chosenOH
+        output << dut.io.output
+
+        when(reset || past(reset)) {
+          for (i <- 0 until portCount) {
+            assume(inputs(i).valid === False)
+          }
+        }
+
+        for (i <- 0 until portCount) {
+          inputs(i) >> dut.io.inputs(i)
+          inputs(i).formalAssumesSlave()
+        }
+        output.formalCovers(2)
+
+        assert(select < portCount)
+        assert(select === OHToUInt(selectOH))
+        assert(output.payload === inputs(select).payload)
+
+        val allInputsValidOH = OHMasking.first(Vec(inputs.map(_.valid))).asBits
```

Comment:
Would you please put the common code into a unified function returns a new Area or Composite, so that the difference between each is verbose?